src/nd2/_sdk/latest.pyx

```diff
@@ -73,16 +73,26 @@ cdef class ND2Reader:
             self._fh = Lim_FileOpenForReadUtf8(self.path)
             if not self._fh:
                 raise OSError("Could not open file: %s" % self.path)
-            self._is_open = 1
 
+            try:
+                attrs = self._attributes()
+                if not len(attrs) >= 6:
+                    Lim_FileClose(self._fh)
```

Comment:
If you close _fh here, you need to not close it in the `except Exception` path a few lines down.

Reply:
perhaps... the problem here is that I don't really know exactly which line is throwing the exception since I can't test it locally.  are you definitely able to call `self._attributes()` without error?  If not, then I _would_ need to call it a few lines down. I'll break it into two steps